include/boost/regex/v4/regex_workaround.hpp

```diff
@@ -207,9 +203,11 @@ namespace boost{ namespace re_detail{
       const char *strSource 
    )
    {
-      if(std::strlen(strSource) + std::strlen(strDestination) + 1 > sizeInBytes)
+      std::size_t lenSource = std::strlen(strSource) + 1;
+      std::size_t lenDestination = std::strlen(strDestination);
```

Comment:
lenSource and lenDestination names are confusingly similar while one includes the terminating zero and the other doesn't.